### ISSUE TITLE
Regression: Does not install repo's appstream icons since move to appstream 1.x

### ISSUE THREAD

**Issue by mcrha** [PackageKit/PackageKit#809]
Since the https://github.com/PackageKit/PackageKit/issues/670 , the repo's `appstream-icons` data is not installed, thus repos providing their apps with the icons do not show correct app icons.

It's because the `as_utils_install_metadata_file` is backwards incompatible with the `as_utils_install_filename`, it cannot install the `appstream-icons.tar` file. Furthermore, the `as_utils_install_metadata_file` expects the file to be `tar.gz` and to contain the icon size in the name, where also the content of the tar.gz changed.

Example repository: https://download1.rpmfusion.org/nonfree/fedora/steam/41/x86_64/

**Comment by mcrha** [comment]
I forgot to mention, the appstream 1.x has very different expectations about the icons and the file names, it's one file per icon size, where the icon sizes are internal to the `as_utils_install_metadata_file` function. Thus the metadata `type` should be also different (the PackageKit checks for an `appstream` and an `appstream-icons` only right now).

**Comment by ximion** [comment]
This issue ws fixed via https://github.com/ximion/appstream/pull/695, right?
Please reopen it if the problem still persists!

**Comment by mcrha** [comment]
> This issue ws fixed via https://github.com/ximion/appstream/pull/695, right?

Nope, not really. That one is about ZSTD support. This one is about different filename and content of the appstream-icons.* files.

Nonetheless, a better choice would be to fix this on the ximion/appstream side, not in each and every user of the ximion/appstream library. I opened https://github.com/ximion/appstream/issues/709 .
